navit/navigation.c

```diff
@@ -966,17 +966,17 @@ round_distance(int dist)
 }
 
 /** @brief Returns the last element of the simplified numbers containing a distance value.
-*   @return value with the highest distance.
-*/
+ *   @return value with the highest distance.
+ */
 static int
 distance_set_last(void)
 {
 	static int i=0;
-	if (i == 0) {
```

Comment:
Attention! Do not remove this, as **i** is a **static** variable and therefore will keep it's value during multiple calls. You can not expect this to be not zero, as it will only be initialized once.

Having a static internal variable called **i** is indeed not good style, and somebody should definitly have a look at this to clarify why **i** is static. But one should not remove the check until done so.